packages/caip-stream/src/tests/streams.ts

```diff
@@ -0,0 +1,19 @@
+import { createDeferredPromise } from '@metamask/utils';
+import type { Duplex } from 'readable-stream';
+
+export const writeToStream = async (stream: Duplex, message: unknown) => {
+  const { promise: isWritten, resolve: writeCallback } =
+    createDeferredPromise();
+
+  stream.write(message, () => writeCallback());
+  await isWritten;
+};
+
+export const onData = (stream: Duplex): unknown[] => {
```

Comment:
Nit: This does listen to the `data` event on `stream`, but that seems to be an implementation detail. The purpose seems to be to collect the chunks from the stream. So, what about:

``` suggestion
export const getChunks = (stream: Duplex): unknown[] => {
```

or perhaps, as an opposite of `writeToStream`:

``` suggestion
export const readFromStream = (stream: Duplex): unknown[] => {
```